packages/plugin-dom/DomParsingEngine.ts

```diff
@@ -10,9 +10,13 @@ export class DomParsingEngine extends ParsingEngine<Node> {
      * @param node
      */
     parseAttributes(node: Element): Record<string, string> {
-        return Array.from(node.attributes).reduce((attributes, attribute) => {
-            attributes[attribute.name] = attribute.value;
-            return attributes;
-        }, {});
+        if (node.attributes) {
+            return Array.from(node.attributes).reduce((attributes, attribute) => {
+                attributes[attribute.name] = attribute.value;
+                return attributes;
+            }, {});
+        } else {
+            return {};
+        }
```

Comment:
What is the difference ?

Reply:
We only read the node's attributes if it has any.
In retrospect I could have done `Array.from(node.attributes || []).reduce` instead. Would you have preferred that?